app/src/main/kotlin/com/hippo/ehviewer/ui/main/DownloadCard.kt

```diff
@@ -0,0 +1,279 @@
+                    }.fillMaxWidth(),
+                    overflow = TextOverflow.Ellipsis,
+                    style = MaterialTheme.typography.titleSmall,
+                )
+                if (downloadState != DownloadInfo.STATE_DOWNLOAD) {
+                    ProvideTextStyle(MaterialTheme.typography.labelLarge) {
+                        info.uploader?.let {
+                            Text(
+                                text = it,
+                                modifier = Modifier.constrainAs(uploaderRef) {
+                                    start.linkTo(parent.start)
+                                    bottom.linkTo(ratingRef.top)
+                                }.alpha(if (info.disowned) 0.5f else 1f),
+                                maxLines = 1,
+                                overflow = TextOverflow.Ellipsis,
+                            )
+                        }
+                        GalleryListCardRating(
+                            rating = info.rating,
+                            modifier = Modifier.constrainAs(ratingRef) {
+                                start.linkTo(parent.start)
+                                bottom.linkTo(categoryRef.top)
+                            },
+                        )
+                        val categoryColor = EhUtils.getCategoryColor(info.category)
+                        val categoryText = EhUtils.getCategory(info.category).uppercase()
+                        Text(
+                            text = categoryText,
+                            modifier = Modifier.constrainAs(categoryRef) {
+                                start.linkTo(parent.start)
+                                bottom.linkTo(parent.bottom)
+                            }.clip(ShapeDefaults.Small).background(categoryColor).padding(vertical = 2.dp, horizontal = 8.dp),
+                            color = if (Settings.harmonizeCategoryColor) Color.Unspecified else EhUtils.categoryTextColor,
+                            textAlign = TextAlign.Center,
+                        )
+                    }
+                } else {
+                    // TODO: use JvmInline Value Class
+                    val updated by DownloadManager.updatedDownloadInfo(info) {
+                        Triple(total, finished, speed)
+                    }
+                    val (total, finished, speed) = updated
+                    ProvideTextStyle(MaterialTheme.typography.labelMedium) {
+                        if (total <= 0 || finished < 0) {
+                            LinearProgressIndicator(
+                                modifier = Modifier.constrainAs(progressBarRef) {
+                                    bottom.linkTo(actionsRef.top)
+                                },
+                            )
+                        } else {
+                            LinearProgressIndicator(
+                                progress = { total.toFloat() / finished },
```

Comment:
Typo?

Reply:
LMAO, confusing why it does not work for a while